### ISSUE TITLE
Sik-Ho Tang | [Paper] Dropout: A Simple Way to Prevent Neural Networks from Overfitting (Image Classification).

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#84]
Sik-Ho Tang. [[Paper] Dropout: A Simple Way to Prevent Neural Networks from Overfitting (Image Classification)](https://sh-tsang.medium.com/paper-dropout-a-simple-way-to-prevent-neural-networks-from-overfitting-image-classification-a74b369b4b8e).

**Comment by NorbertZheng** [comment]
## Overview
A Very Famous Regularization Approach to Prevents Co-Adaptation so as to Reduce Overfitting.

Inthis paper, Dropout: A Simple Way to Prevent Neural Networks from Overfitting (Dropout), by University of Toronto, is shortly presented.
 - The key idea is to randomly drop units (along with their connections) from the neural network during training.
 - This prevents units from co-adapting too much.

This is firstly appeared in 2012 arXiv with over 5000 citations. And it is used in [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160?source=post_page---------------------------) (2012 NIPS), with over 73000 citations and got the first place in 2012 ImageNet competition. Finally, it is published in 2014 JMLR with over 23000 citations.

**Comment by NorbertZheng** [comment]
## Dropout
### General Idea
![image](https://user-images.githubusercontent.com/37235023/236663669-2d075a08-14b5-45f6-87db-ef4a56c80e78.png)
Dropout.

 - Left: When using the neural network at the left, if there are some neurons which are quite strong, the network will depend those neurons too much making others weak and unreliable.
 - Right: An example of a thinned net produced by applying dropout to the network on the left. Crossed units have been dropped.

### Training & Testing
![image](https://user-images.githubusercontent.com/37235023/236663733-2b6963ba-722e-4c41-84c5-ff783ab4b251.png)
 - Left: At training time, each unit has probability $p$ that it will not be used.
 - Right: At test time, the weights of this network are **scaled-down versions of the trained weights**.
 - If a unit is retained with probability $p$ during training, the outgoing weights of that unit are multiplied by $p$ at test time as shown in the above figure.

Leading to redundant disentangled representation???
 - It is conjectured that co-adaptations do not generalize to unseen data.
 - Dropout prevents co-adaptation by making the presence of other hidden units unreliable.

**Comment by NorbertZheng** [comment]
## Experimental Results
### MNIST
![image](https://user-images.githubusercontent.com/37235023/236663895-583e3cb9-5a89-4542-a834-0704c22538e9.png)
Error Rate on MNIST.

 - All dropout nets use $p=0.5$ for hidden units and $p=0.8$ for input units.
 - The best-performing neural networks without dropout obtains 1.6% error rate. With dropout, the error reduces to 1.35%.
 - With other kinds of improvement such as using ReLU, more neurons, **max-norm constraint**, [**Maxout**](https://towardsdatascience.com/review-maxout-network-image-classification-40ecd77f7ce4?source=post_page---------------------------), 0.94% error rate is achieved.

![image](https://user-images.githubusercontent.com/37235023/236663972-49aa8226-3ad4-425b-b229-b48654bcc64c.png)
Test error for different architectures with and without dropout.

 - The same architectures trained with and without dropout have drastically different test errors as seen as by the two separate clusters of trajectories. Dropout gives a huge improvement across all architectures.

**Comment by NorbertZheng** [comment]
### Street View House Numbers (SVHN)
![image](https://user-images.githubusercontent.com/37235023/236664110-56e29c9e-b641-4990-8ed7-fb21d01c6b78.png)
Error Rates on Street View House Numbers.

 - The best architecture is [LeNet](https://medium.com/@sh.tsang/paper-brief-review-of-lenet-1-lenet-4-lenet-5-boosted-lenet-4-image-classification-1f5f809dbf17?source=post_page---------------------------) which has three convolutional layers followed by 2 fully connected hidden layers. All hidden units were ReLUs.
 - The best-performing convolutional nets that do not use dropout achieve an error rate of 3.95%.
 - Adding dropout only to the fully connected layers reduces the error to 3.02%.
 - Adding dropout to the convolutional layers as well further reduces the error to 2.55%.

**Dropout in the lower layers still helps because it provides noisy inputs for the higher fully connected layers which prevents them from overfitting.**
 - Even more gains can be obtained by using [Maxout](https://towardsdatascience.com/review-maxout-network-image-classification-40ecd77f7ce4?source=post_page---------------------------) units.

**Comment by NorbertZheng** [comment]
### CIFAR
![image](https://user-images.githubusercontent.com/37235023/236664242-df43e158-3f5b-4e70-a6be-a2c8702552e9.png)
Error rates on CIFAR-10 and CIFAR-100.

 - The best Conv Net without dropout obtains an error rate of 14.98% on CIFAR-10.
 - Using dropout in the fully connected layers reduces that to 14.32% and adding dropout in every layer further reduces the error to 12.61%.
 - Error is further reduced to 11.68% by replacing ReLU units with [Maxout](https://towardsdatascience.com/review-maxout-network-image-classification-40ecd77f7ce4?source=post_page---------------------------) units.
 - On CIFAR-100, dropout reduces the error from 43.48% to 37.20% which is a huge improvement.

**Comment by NorbertZheng** [comment]
### ImageNet
![image](https://user-images.githubusercontent.com/37235023/236664301-f391da70-eb28-4b8a-bcc7-8d6327359e4f.png)
Top-1 and Top-5 error rates on ImageNet validation and test sets.

 - [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160?source=post_page---------------------------) based on convolutional nets and dropout won the ILSVRC-2012 competition.
 - While the best methods based on standard vision features achieve a top-5 error rate of about 26%, convolutional nets with dropout achieve a test error of about 16%.

**Comment by NorbertZheng** [comment]
There are also results for speech and test datasets, as well as detailed studies and discussions about Dropout. Please feel free to read the paper especially 2014 JMLR one.

(I’ve just felt that it’s seems missing something if I write stories about other regularization methods later on, though I think many people have already known about Dropout…)

**Comment by NorbertZheng** [comment]
## References
 - [2012 arXiv] [Dropout] [Improving neural networks by preventing co-adaptation of feature detectors](https://arxiv.org/abs/1207.0580).
 - [2014 JMLR] [Dropout] [Dropout: A Simple Way to Prevent Neural Networks from Overfitting](https://jmlr.org/papers/v15/srivastava14a.html).
